python/tvm/driver/tvmc/autotuner.py

```diff
@@ -293,9 +298,66 @@ def drive_tune(args):
         include_simple_tasks=args.include_simple_tasks,
         log_estimated_latency=args.log_estimated_latency,
         additional_target_options=reconstruct_target_args(args),
+        tasks_filter=args.tasks,
     )
 
 
+def filter_tasks(
+    tasks: Optional[Union[auto_scheduler.SearchTask, autotvm.task.Task]],
+    expr: str,
+):
+    assert isinstance(expr, str), "Expected filter expression of string type"
+    assert len(expr) > 0, "Got empty filter expression"
+
+    # groups of keywords are comma-separated
+    splitted = expr.split(",")
+
+    do_list = False
+    do_filter = False
+    selected = []
+    for item in splitted:
+        if item in ["list", "help"]:
+            do_list = True
+        elif item in ["all"]:
+            selected = list(range(len(tasks)))
+        else:
+            do_filter = True
+            if "-" in item:
+                lhs, rhs = item.split("-")[:2]
+                if len(lhs) == 0:
```

Comment:
i think you could maybe make this a bit more concise:
lhs = 
```suggestion
                lhs = int(lhs) if lhs else 0
                rhs = int(rhs) if rhs else len(tasks) - 1
                assert 0 <= lhs < len(tasks), "explanation"
                assert 0 <= rhs < len(tasks), "explanation"
```